### ISSUE TITLE
Relation with MVP

### ISSUE THREAD

**Issue by Konrad-Morawski** [konmik/Phlux#9]
Hello Konstantin,
I'm trying to wrap my head around this pattern. 
I think I get the gist of it: I appreciate the benefits of immutability and purity, how it simplifies tracking changes, it makes sense. 
I'm also fairly familiar with MVP on Android.

I don't really get why they're mutually exclusive though - so apparently something escapes me.

In MVP, the Presenter is notified about user actions (detected on UI level), reacts to them by notifying the Model and propagates the results back to the layout. 

But it can very well propagate them back by dispatching Reducer actions - right? Why not?

Basically I fail to see why would both these approaches rule eachother out rather than complement. Is it semantics? Doesn't Store become an alias name for the Presenter? What is it that I'm missing?

Thanks for any insights / clarifications

**Comment by konmik** [comment]
Hi, the difference in fact is between OOP and FP.

In OOP you have an *entity* (collection of mutable variables) that need to be notified, react, fight multithreading issues, fight with order of variable changes, fight with lifecycles, etc. There is a lots of "accidental complexities" with this approach. Your app does not need them, why do you fight with them then?

FP is the main cure against such complexities. Instead of fighting with complexities, FP removes them.

So, mixing MVP with Flux is the same as mixing OOP with FP. You get half of your app cured, but...
